[Diagnostics]: Bad diagnostic for initialization of property wrapper with custom init

saehejkang opened the pull request:
<!-- What's in this pull request? -->

Updates error message for initialization of property wrapper with a custom `init`.

<!--
If this pull request resolves any GitHub issues, link them.
For information about linking a pull request to an issue, see:
https://docs.github.com/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue
-->
Resolves #69066

<!--
Before merging this pull request, you must run the Swift continuous integration tests.
For information about triggering CI builds via @swift-ci, see:
https://github.com/apple/swift/blob/main/docs/ContinuousIntegration.md#swift-ci

Thank you for your contribution to Swift!
-->

saehejkang committed:
[Diagnostics]: update error message + test

saehejkang commented:
I am aware all the test cases for `SILOptimizer/di_property_wrappers_errors.swift` are failing now and am looking into it.

AnthonyLatsis commented:
The error message you modified is used to diagnose argument label mismatches in general: 
```swift
func foo(x: Int) {}
foo(y: 0)
```

saehejkang commented:
> The error message you modified is used to diagnose argument label mismatches in general

I simply searched for instances of the wrong error message and updated it accordingly. Is there an issue with the message I updated or are you saying that I wrote the test in the wrong place?

saehejkang commented:
I am no longer working on this issue
